### DISCUSSION TITLE
[Feature] I would like to use cn clip_ Vit-b-16, how can I switch or import this Chinese CLIP model?

### DISCUSSION THREAD

**Discussion by nodis** [immich-app/immich#4485] [Feature Request]
### The feature

I would like to use cn clip_ Vit-b-16, how can I switch or import this Chinese CLIP model? thanks

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
Hi! We don't support CN CLIP models at the moment, but we do support several multilingual models like `M-CLIP/XLM-Roberta-Large-Vit-B-32`. See [this](https://clip-as-service.jina.ai/user-guides/benchmark/#size-and-efficiency) list for other options (look for model names that start with "M-CLIP").

**Reply by nodis** [reply]
Thank you, I have seen many Chinese CLIP models. Do you have a tutorial on how to integrate Chinese CLIP models into immich?

**Reply by mertalev** [reply]
Using custom models outside of that list isn't currently supported.

**Comment by si9ma** [comment]
I tried to change it to support cn clip, you can pull and try if you are interested https://github.com/si9ma/immich
我尝试改了一下，有兴趣的可以pull了试一下，效果还可以 https://github.com/si9ma/immich @nodis 
<img width="2581" alt="image" src="https://github.com/immich-app/immich/assets/24809310/acf17938-1e08-4e49-976f-b7c411b4eae3">
